### PR TITLE
Ab/go eh update

### PR DESCRIPTION
**Pull request by aarohib**
## What does this PR do?
Corrects info on the [Error Handling in Go](https://docs.temporal.io/docs/go/error-handling) page.

### CODE REVIEW

File: docs/go/how-to-test-workflow-definitions-in-go.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ---
-id: testing
-title: Testing and debugging
+id: how-to-test-workflow-definitions-in-go
+title: How to test Workflow defitions in Go
```
Comment (flossypurse) [inline docs/go/how-to-test-workflow-definitions-in-go.md:3]:
```suggestion
title: How to test Workflow Definitions in Go
```

**Review by flossypurse — COMMENTED** [review]
This could be a good opportunity to do a little copy editing to the page itself - if you want.
Or at minimum you could run through it and put every sentence on its own line kind of thing.

Don't want to open a can of worms, so up to you @aarohib -

File: blog/2021-09-14-build-an-ecommerce-app-with-temporal-part-4.md [hunk 2]
```diff
@@ -22,7 +22,7 @@ import TabItem from '@theme/TabItem';
 In [Part 1](/blog/build-an-ecommerce-app-with-temporal-part-1/), [Part 2](/blog/build-an-ecommerce-app-with-temporal-part-2-reminder-emails/), and [Part 3](/blog/build-an-ecommerce-app-with-temporal-part-3-testing), you built and tested a shopping cart with an abandoned cart email notification using long-lived Workflows.
 Workflows, Activities, and Temporal's testing utilities make it easy to build and maintain features that involve external services and time, like sending an email reminder when a user hasn't touched their cart in a while.
 
-Thus far, you've only worked with the Temporal SDK via [starters](/docs/go/hello-world-tutorial/#workflow-starter) and [unit tests](/docs/go/testing), which invoke the Temporal SDK directly.
+Thus far, you've only worked with the Temporal SDK via [starters](/docs/go/hello-world-tutorial/#workflow-starter) and [unit tests](/docs/go/how-to-test-workflow-definitions-in-go), which invoke the Temporal SDK directly.
```
Comment (djmagee) [inline blog/2021-09-14-build-an-ecommerce-app-with-temporal-part-4.md:25]:
```suggestion
Thus far, you've worked only with the Temporal SDK via [starters](/docs/go/hello-world-tutorial/#workflow-starter) and [unit tests](/docs/go/how-to-test-workflow-definitions-in-go), which invoke the Temporal SDK directly.
```

File: docs/go/error-handling.md [hunk 3]
```diff
@@ -7,7 +7,7 @@ An Activity, or a child Workflow, might fail, and you could handle errors differ
 error cases.
 
 If the Activity returns an error as `errors.New()` or `fmt.Errorf()`, that error will
```
Comment (djmagee) [inline docs/go/error-handling.md:9]:
```suggestion
If the Activity returns an error as `errors.New()` or `fmt.Errorf()`, that error is
```

File: docs/go/error-handling.md [hunk 4]
```diff
@@ -7,7 +7,7 @@ An Activity, or a child Workflow, might fail, and you could handle errors differ
...
-be converted to `*temporal.ApplicationError` and wrapped inside `*temporal.ActivityTaskError` or `*temporal.ChildWorkflowExecutionError`.
+be converted to `*temporal.ApplicationError`.
```
Comment (djmagee) [inline docs/go/error-handling.md:10]:
```suggestion
converted to `*temporal.ApplicationError`.
```

File: docs/go/how-to-test-workflow-definitions-in-go.md (hunk 1 above)
Comment (aarohib) [reply docs/go/how-to-test-workflow-definitions-in-go.md]: Updated. Fixed some other typos as well.

File: docs/go/error-handling.md (hunk 3 above)
Comment (aarohib) [reply docs/go/error-handling.md:9]: Done

File: docs/go/error-handling.md (hunk 4 above)
Comment (aarohib) [reply docs/go/error-handling.md:10]: DOne

File: blog/2021-09-14-build-an-ecommerce-app-with-temporal-part-4.md (hunk 2 above)
Comment (aarohib) [reply blog/2021-09-14-build-an-ecommerce-app-with-temporal-part-4.md]: Done

File: docs/go/how-to-test-workflow-definitions-in-go.md [hunk 5]
```diff
@@ -1,13 +1,14 @@
 ---
 id: how-to-test-workflow-definitions-in-go
-title: How to test Workflow defitions in Go
+title: How to test Workflow definitions in Go
```
Comment (flossypurse) [inline docs/go/how-to-test-workflow-definitions-in-go.md:3]:
```suggestion
title: How to test Workflow Definitions in Go
```

File: docs/go/error-handling.md [hunk 6]
```diff
@@ -3,13 +3,13 @@ id: error-handling
...
 ---
 
-An Activity, or a child Workflow, might fail, and you could handle errors differently based on the different
+An Activity, or a Child Workflow, might fail, and you could handle errors differently based on the different
 error cases.
 
-If the Activity returns an error as `errors.New()` or `fmt.Errorf()`, that error will
-be converted to `*temporal.ApplicationError` and wrapped inside `*temporal.ActivityTaskError` or `*temporal.ChildWorkflowExecutionError`.
+If the Activity returns an error as `errors.New()` or `fmt.Errorf()`, that error is
+converted to `*temporal.ApplicationError`.
```
Comment (flossypurse) [inline docs/go/error-handling.md:10]:
```suggestion
If the Activity returns an error as `errors.New()` or `fmt.Errorf()`, that error is converted into `*temporal.ApplicationError`.
```

**Review by flossypurse — APPROVED** [review]

**Pull request merged**